crypto/nrf_cc310_platform/src/nrf_cc3xx_platform_mutex_freertos.c

```diff
@@ -103,7 +103,7 @@ static void mutex_init_platform(nrf_cc3xx_platform_mutex_t *mutex)
         platform_abort_apis.abort_fn("mutex_init called with NULL parameter");
     }
 
-    if (mutex->flag == NRF_CC3XX_PLATFORM_MUTEX_MASK_INVALID)
+    if (mutex->flags == NRF_CC3XX_PLATFORM_MUTEX_MASK_INVALID)
```

Comment:
Do you not need to OR-out the flag you are interested in?

It is called flags but there can only be one?

Perhaps a comment explaining why it is safe to do this thing that is usually dangerous. Or add an OR in case there is a risk of multiple flags in the future.

Reply:
I just fixed a typo here to be honest, didn't change any functionality. But in any case I think that this is correct. When the when the mask is invalid is when you actually need to create an RTOS mutex. If any other flag is added/set then you definitely don't need to create a mutex. But I don't think that it's very relevant to the patch, I can easily revert the change if needed. I will not change the code in the freertos code (functionality) right now since I don't have the capacity to test it for this release.

Reply:
Ah, I misunderstood.

I thought this was a porting patch.

I thought that it used to be flag, and now is flags, thereby introducing the risk of incorrect masking.

If this was a typo then the patch is sane.